andela/ah-bird-box-frontend
# #163388217 CRUD for comments

#### What does this PR do?
Have CRUD for comments

#### Description of Task to be completed?

- add create comment
- add edit comment
- add fetch all comments
- add reply comment
- add delete comment

#### How should this be manually tested?
- Clone the repo and cd into it.

- Checkout to branch `git checkout users-comment-on-articles-163388217`

- `npm install` to install dependancies

- `npm start` to start server

- Login to the application

- Click on any article to view the article.

- Add a comment to the article to view a comment you've created

- After viewing, you can edit the comment by clicking on the edit icon

- To delete your comment, click on the delete icon

- To reply on a comment, click  on reply. 

- Click on the check box to view replies

**NB:** You can only view the edit and delete icon if you are the user that created the comment

#### Any background context you want to provide?
None



#### What are the relevant pivotal tracker stories?
[#163388217](https://www.pivotaltracker.com/n/projects/2240063/stories/163388217)


#### Screenshots (if appropriate):
View all comments
![image](https://user-images.githubusercontent.com/42655718/55214019-5cb29680-5206-11e9-8009-ce89dbe776a9.png)

Add a comment 

![image](https://user-images.githubusercontent.com/42655718/55214047-7e138280-5206-11e9-8e99-d23703a52704.png)

Update a comment 
![image](https://user-images.githubusercontent.com/42655718/55214053-88ce1780-5206-11e9-9454-9b6e66515653.png)

Reply to a comment 
![image](https://user-images.githubusercontent.com/42655718/55214083-9f746e80-5206-11e9-853b-735dee7e936c.png)

Delete a comment
![image](https://user-images.githubusercontent.com/42655718/55214090-a9966d00-5206-11e9-82a0-a84b93d5d3d9.png)


#### Questions:

None

#### Reviewers:

@TooColline
@calebrotich10
@unah254
@bryan-munene
@hoslack
